jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -101,10 +101,14 @@ public Throwable getThrowable() {
 	
 	private static native boolean isCheckpointAllowed();
 
-	private static native CRIUResult checkpointJVMImpl(String checkPointDir, boolean keepRunning, boolean shellJob, boolean extUnixSupport, int logLevel, String logFile);
+	private native CRIUResult checkpointJVMImpl();
 
 	private CRIUSupport() {}
 
+	public static CRIUSupport newCRIUSupport() {
+		return new CRIUSupport();
+	}
```

Comment:
Why not just make the constructor public?

Reply:
I think it is because we want usage like:
```java
if (CRIUSupport.isCRIUSupportEnabled()) {
    try {
          CRIUSupport.newCRIUSupport()
          .setImagesDir(checkpointPath)
          .setLogLevel(logPath)
          .checkPointJVM();
    } catch (UnsupportedOperation e) {
         //can get here if isCRIUSupportEnabled is false
         //or if running in mode where a restored JVM is not allowed to checkpoint
    } catch (CheckPointException e) {
         //can get here if there is a JVM or CRIU failure when attempting to checkpoint
         //exception message will have relevant details
    } catch (RestoreException e) {
         //Can get here is there is a failure when restoring the JVM. 
         //exception message has relevant details
         //Note failures here are non-fatal
         //If a fatal failure occurs the JVM process exits
    }
}
```

Reply:
Replacing the first line
```
  CRIUSupport.newCRIUSupport()
```
with
```
  new CRIUSupport()
```
the rest of the pattern continues to work.